app/src/lib/editors/win32.ts

```diff
@@ -10,663 +10,595 @@ import {
+      let displayName = ''
+      let publisher = ''
+      let installLocation = ''
+
+      for (const item of keys) {
+        // NOTE:
+        // Webstorm adds the current release number to the end of the Display Name, below checks for "PhpStorm"
+        if (
+          item.name === 'DisplayName' &&
+          item.type === RegistryValueType.REG_SZ &&
+          item.data.startsWith('PhpStorm ')
+        ) {
+          displayName = 'PhpStorm'
+        } else if (
+          item.name === 'Publisher' &&
+          item.type === RegistryValueType.REG_SZ
+        ) {
+          publisher = item.data
+        } else if (
+          item.name === 'InstallLocation' &&
+          item.type === RegistryValueType.REG_SZ
+        ) {
+          installLocation = item.data
+        }
       }
-    }
-
-    return { displayName, publisher, installLocation }
-  }
-
-  if (editor === ExternalEditor.NotepadPlusPlus) {
-    const displayName = getKeyOrEmpty(keys, 'DisplayName')
-    const publisher = getKeyOrEmpty(keys, 'Publisher')
-    const installLocation = getKeyOrEmpty(keys, 'DisplayIcon')
 
-    return { displayName, publisher, installLocation }
-  }
-
-  if (editor === ExternalEditor.Rider) {
-    let displayName = ''
-    let publisher = ''
-    let installLocation = ''
-
-    for (const item of keys) {
-      // NOTE:
-      // JetBrains Rider adds the current release number to the end of the Display Name, below checks for "JetBrains Rider"
-      if (
-        item.name === 'DisplayName' &&
-        item.type === RegistryValueType.REG_SZ &&
-        item.data.startsWith('JetBrains Rider ')
-      ) {
-        displayName = 'JetBrains Rider'
-      } else if (
-        item.name === 'Publisher' &&
-        item.type === RegistryValueType.REG_SZ
-      ) {
-        publisher = item.data
-      } else if (
-        item.name === 'InstallLocation' &&
-        item.type === RegistryValueType.REG_SZ
-      ) {
-        installLocation = item.data
+      return { displayName, publisher, installLocation }
+    },
+    expectedInstallationChecker: (displayName, publisher) =>
+      displayName.startsWith('PhpStorm') && publisher === 'JetBrains s.r.o.',
+  },
+  {
+    name: 'Notepad++',
+    registryKeys: [
+      // 64-bit version of Notepad++
+      {
+        key: HKEY.HKEY_LOCAL_MACHINE,
+        subKey:
+          'SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Notepad++',
+      },
+      // 32-bit version of Notepad++
+      {
+        key: HKEY.HKEY_LOCAL_MACHINE,
+        subKey:
+          'SOFTWARE\\WOW6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Notepad++',
+      },
+    ],
+    executableShimPath: [],
+    appInformationExtractor: keys => {
+      const displayName = getKeyOrEmpty(keys, 'DisplayName')
+      const publisher = getKeyOrEmpty(keys, 'Publisher')
+      const installLocation = getKeyOrEmpty(keys, 'DisplayIcon')
+
+      return { displayName, publisher, installLocation }
+    },
```

Comment:
If we leverage the default extractor here we can make it clear that we're only overriding the `installLocation`
```suggestion
    appInformationExtractor: keys => ({
      ...defaultAppInformationExtractor(keys),
      installLocation: getKeyOrEmpty(keys, 'DisplayIcon'),
    }),
```